### PR TITLE
EVA - 3121 Ensure all samples are from SARS-COV-2

### PR DESCRIPTION
**Pull request by tcezard**
This PR add 2 feature to skip specific samples:
 - First all analysis not marked with the SARS-COV-2 taxonomy id will be skipped and stored in a separate `ignored-analyses-file`
 - Any other analysis added to this `ignored-analyses-file` will be skipped regardless of their taxonomy.

### CODE REVIEW

File: tests/test_download_analyses.py [hunk 1]
```diff
@@ -17,12 +17,16 @@ class TestDownloadSnapshot(TestCase):
     toplevel_download_folder = os.path.join(resources_folder, 'download_analyses')
     download_target_dir = os.path.join(toplevel_download_folder, '30_eva_valid')
     processed_analyses_file = os.path.join(toplevel_download_folder, 'processed_analyses_file.txt')
+    ignored_analyses_file = os.path.join(toplevel_download_folder, 'ignored_analyses_file.txt')
     project = 'PRJEB45554'
+    accepted_taxonomies = [2697049]
```
Comment (apriltuesday) [inline tests/test_download_analyses.py:87]: Is it worth adding a test for this (i.e. if `accepted_taxonomies` is something completely different, nothing gets downloaded)?

File: covid19dp_submission/download_analyses.py [hunk 2]
```diff
@@ -67,27 +68,43 @@ def total_analyses_in_project(project):
...
+        # Filter out based on previously marked analysis
+        unprocessed_analyses = filter_out_processed_analyses(analyses_from_ena, analysis_to_skip)
+        # Filter out based on taxonomy
+        unprocessed_analyses = [a for a in unprocessed_analyses if int(a.get('tax_id') or 0) in accepted_taxonomies]
+        # Gather filtered analysis that had not been previously filtered out
+        new_files_to_ignore.extend([
+            a for a in analyses_from_ena
+            if int(a.get('tax_id') or 0) not in accepted_taxonomies and
+               a.get('analysis_accession') not in analysis_to_ignore
+        ])
```
Comment (apriltuesday) [inline covid19dp_submission/download_analyses.py]: This includes analyses that have already been processed with unacceptable taxonomies, right?  Is that intended?  (I guess it only matters for the first run after this fix is deployed, at which point these all get added to the ignored analysis file anyway.)

Personally I would find it more readable for this code to just split the unprocessed analyses into ones with accepted taxonomies and ones without, up to you though.

**Review by apriltuesday — APPROVED** [review]

File: tests/test_download_analyses.py (hunk 1 above)
Comment (tcezard) [reply tests/test_download_analyses.py:87]: Added.

File: covid19dp_submission/download_analyses.py (hunk 2 above)
Comment (tcezard) [reply covid19dp_submission/download_analyses.py]: I've changed this part to make it more readable. It's not perfect but I think it should do for now.
I was checking all the previously processed variant to identified others that might have the wrong taxonomies (and found some). This said, you're right that it should not be done every time.

**Review by apriltuesday — APPROVED** [review]
Thanks for the tests, this looks good to me!

**Review by sundarvenkata-EBI — APPROVED** [review]

**Pull request merged**